### PR TITLE
fix: Remove 'label' attribute from <label> tag

### PR DESCRIPTION
**Pull request by abdulbasithqb**
#### What does this PR do?
fix: Remove 'label' attribute from <label> tag

#### Where should the reviewer start?
CheckBox custom element

#### What testing has been done on this PR?
CheckBox custom element

#### How should this be manually tested?
1. Local Jest tests.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
No

#### What are the relevant issues?
Fixes: #7096

#### Screenshots (if appropriate)
Screenshot 2024-01-29 at 2 23 23 PM

#### Is this change backwards compatible or is it a breaking change?
 No

### CODE REVIEW

File: src/js/components/CheckBox/CheckBox.js [hunk 1]
```diff
@@ -215,7 +215,6 @@ const CheckBox = forwardRef(
         reverse={reverse}
         {...removeUndefined({ htmlFor: id, disabled })}
         checked={checked}
-        label={label}
```
Comment (britt6612) [inline src/js/components/CheckBox/CheckBox.js:218]: can we add this back in and maybe have `labelProp={label}`

File: src/js/components/CheckBox/__tests__/__snapshots__/CheckBox-test.tsx.snap [hunk 2]
```diff
@@ -225,10 +225,6 @@ exports[`CheckBox controlled 1`] = `
   -webkit-flex-direction: row;
   -ms-flex-direction: row;
   flex-direction: row;
-  -webkit-align-items: center;
```
Comment (britt6612) [inline src/js/components/CheckBox/__tests__/__snapshots__/CheckBox-test.tsx.snap:228]: we need these styles to still get placed on the label so still need `labelProp` defined in checkbox.js

File: src/js/components/CheckBox/CheckBox.js (hunk 1 above)
Comment (britt6612) [reply src/js/components/CheckBox/CheckBox.js:218]: if we change this to `labelProp` we will also need to change 
```
typeof props.label

```
to 
```
typeof props.labelProp

```

https://github.com/grommet/grommet/blob/master/src/js/components/CheckBox/StyledCheckBox.js#L57

File: src/js/components/CheckBox/__tests__/__snapshots__/CheckBox-test.tsx.snap (hunk 2 above)
Comment (abdulbasithqb) [reply src/js/components/CheckBox/__tests__/__snapshots__/CheckBox-test.tsx.snap]: reverted snapshots, please check now

**Review by britt6612 — APPROVED** [review]
LGTM

**Review by jcfilben — APPROVED** [review]
Looks good!

**Pull request merged**